### PR TITLE
[cpptest] Use find_package to locate GTest files

### PR DESCRIPTION
**Pull request by kparzysz-quic**
There is a standard CMake utility for finding GTest, use that instead of doing a manual search.

Thanks @tkonolige for the suggestion!

### CODE REVIEW

**Comment by Mousius** [comment]
Hi @kparzysz-quic, is it possible to do this for `GMock` as well?

**Comment by kparzysz-quic** [comment]
I don't know.  From what I just googled it seems like GMock is somehow handled by GoogleTest, which is included by the FindGTest routine.

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,6 +52,7 @@ tvm_option(INDEX_DEFAULT_I64 "Defaults the index datatype to int64" ON)
 tvm_option(USE_LIBBACKTRACE "Build libbacktrace to supply linenumbers on stack traces" AUTO)
 tvm_option(BUILD_STATIC_RUNTIME "Build static version of libtvm_runtime" OFF)
 tvm_option(USE_PAPI "Use Performance Application Programming Interface (PAPI) to read performance counters" OFF)
+tvm_option(USE_GTEST "Use GoogleTest for C++ sanity tests" AUTO)
```
[the remark this answers is not in the document]
Comment (kparzysz-quic) [reply CMakeLists.txt:55]: Done

File: CMakeLists.txt [hunk 2]
```diff
@@ -585,22 +596,14 @@ endif()
...
 # Create the `cpptest` target if we can find GTest.  If not, we create dummy
 # targets that give the user an informative error message.
-if(GTEST_INCLUDE_DIR AND GTEST_LIB)
+if(GTEST_FOUND)
   file(GLOB_RECURSE TEST_SRCS tests/cpp/*.cc)
   add_executable(cpptest ${TEST_SRCS})
-  target_include_directories(cpptest SYSTEM PUBLIC ${GTEST_INCLUDE_DIR})
-  target_link_libraries(cpptest PRIVATE ${TVM_TEST_LIBRARY_NAME} ${GTEST_LIB} gtest_main pthread dl)
+  target_include_directories(cpptest SYSTEM PUBLIC ${GTEST_INCLUDE_DIRS})
+  target_link_libraries(cpptest PRIVATE ${TVM_TEST_LIBRARY_NAME} ${GTEST_BOTH_LIBRARIES} pthread dl)
```
[the remark this answers is not in the document]
Comment (kparzysz-quic) [reply CMakeLists.txt]: Done.

**Comment by tkonolige** [comment]
@Mousius If we required a recent version of gmock, then we could just use the cmake config it provides. But all versions of Ubuntu seem to not include it, so that approach is out. So unfortunately I think you'll have to write your own `FindGMock` cmake file. I find it easiest to use [pkg_search_module](https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_search_module) which uses PKGConfig to find the libraries.

**Comment by areusch** [comment]
seems CI is green, let's land this and we can always do the GMock in a follow-on.

**Pull request merged**